### PR TITLE
Declared functions in Borehole package as pure

### LINKED ISSUE #1582: Incorrectly tagged impure functions in geothermal package
`IBPSA.Fluid.Geothermal.Borefields.BaseClasses.Boreholes.BaseClasses.Functions.internalResistancesOneUTube` and `IBPSA.Fluid.Geothermal.Borefields.BaseClasses.Boreholes.BaseClasses.Functions.internalResistancesTwoUTube` have had the impure keyword introduced in #1564.

I am not sure these functions require the `impure` keyword. Their outputs are predictable, and do not rely on read/write operations on external files. They may have been mixed in with `IBPSA.Fluid.Geothermal.Borefields.BaseClasses.HeatTransfer.LoadAggregation.temperatureResponseMatrix` due to the use of a variable named `sha`.

The contexts are different, in the first case `sha` is the shank spacing and in the second `sha` is the SHA-1 encryption of g-function arguments used to read/write a file on disk.

### PR DESCRIPTION
**Pull request by mwetter**
This closes #1582

### CODE REVIEW

File: IBPSA/Fluid/Geothermal/Borefields/BaseClasses/Boreholes/BaseClasses/Functions/internalResistancesOneUTube.mo [hunk 1]
```diff
@@ -1,5 +1,5 @@
 within IBPSA.Fluid.Geothermal.Borefields.BaseClasses.Boreholes.BaseClasses.Functions;
-impure function internalResistancesOneUTube
+pure function internalResistancesOneUTube
```
Comment (MassimoCimmino) [inline IBPSA/Fluid/Geothermal/Borefields/BaseClasses/Boreholes/BaseClasses/Functions/internalResistancesOneUTube.mo:2]: Is the `pure` keyword required?

File: IBPSA/Fluid/Geothermal/Borefields/BaseClasses/Boreholes/BaseClasses/Functions/internalResistancesTwoUTube.mo [hunk 2]
```diff
@@ -1,5 +1,5 @@
 within IBPSA.Fluid.Geothermal.Borefields.BaseClasses.Boreholes.BaseClasses.Functions;
-impure function internalResistancesTwoUTube
+pure function internalResistancesTwoUTube
```
Comment (MassimoCimmino) [inline IBPSA/Fluid/Geothermal/Borefields/BaseClasses/Boreholes/BaseClasses/Functions/internalResistancesTwoUTube.mo:2]: Is the `pure` keyword required?

File: IBPSA/Fluid/Geothermal/Borefields/BaseClasses/Boreholes/BaseClasses/Functions/internalResistancesOneUTube.mo (hunk 1 above)
Comment (mwetter) [reply IBPSA/Fluid/Geothermal/Borefields/BaseClasses/Boreholes/BaseClasses/Functions/internalResistancesOneUTube.mo]: It is not required as functions are `pure` by default. I removed the keyword.

File: IBPSA/Fluid/Geothermal/Borefields/BaseClasses/Boreholes/BaseClasses/Functions/internalResistancesTwoUTube.mo (hunk 2 above)
Comment (mwetter) [reply IBPSA/Fluid/Geothermal/Borefields/BaseClasses/Boreholes/BaseClasses/Functions/internalResistancesTwoUTube.mo]: It is not required as functions are `pure` by default. I removed the keyword.

**Review by MassimoCimmino — APPROVED** [review]

**Pull request merged**